src/components/Dialog.tsx

```diff
@@ -51,18 +50,43 @@ export default class Dialog extends Component<DialogProps, DialogState> {
   getSourceIDAndPaths = async (
     props: DialogProps,
   ): Promise<Array<SourceProps>> => {
-    const sources = await this.getSources(props);
-    const enabledSources = sources.data.reduce(
-      (result: SourceProps[], source: any) => {
-        if (source.attributes.enabled) {
-          let id = source.id;
-          let name = source.attributes.name;
-          result.push({ id, name });
-        }
-        return result;
-      },
-      [],
-    );
+    let sources,
+      enabledSources: Array<SourceProps> = [];
+
+    try {
+      sources = await this.getSources(props);
+    } catch (error) {
+      console.error(error);
+      return enabledSources;
+    }
+
+    /*
+     * Resolved requests can either return an array
+     * of objects or a single object via the data
+     * top-level field. In the following block, we
+     * will account for both possibilities.
+     */
+    if (Array.isArray(sources.data)) {
+      enabledSources = sources.data.reduce(
+        (result: SourceProps[], source: any) => {
+          // TODO: add more explicit types for source.data
+          if (source.attributes.enabled) {
+            let id = source.id;
+            let name = source.attributes.name;
+            result.push({ id, name });
+          }
+          return result;
+        },
+        [],
+      );
+    } else if (sources) {
+      const source: any = sources.data; // TODO: add more explicit types for source.data
+      if (source.attributes.enabled) {
+        let id = source.id;
+        let name = source.attributes.name;
+        enabledSources.push({ id, name });
+      }
+    }
```

Comment:
Because I can, I'll detail how I would write this to keep the code DRY.

```ts
    /*
     * Resolved requests can either return an array
     * of objects or a single object via the data
     * top-level field. In the following block, we
     * will account for both possibilities.
     */
    const sourcesArray = Array.isArray(sources.data) ? sources.data : [sources.data];
    const enabledSources = sourcesArray.reduce((prev, source: any) => {
        // TODO: add more explicit types for source.data
        if (source.attributes.enabled) {
          const id = source.id;
          const name = source.attributes.name;
          prev.push({ id, name });
        }
        return prev;
      },
      [] as SourceProps[],
    );
```

`enabledSources` no longer needs to be defined above with a `let`, reducing another opportunity for a bug

Reply:
Nicee, yeah that definitely needed some cleaning up